app/Http/Controllers/RelationshipController.php

```diff
@@ -5,22 +5,34 @@
 use Illuminate\Http\Request;
 use App\Relationship;
 use App\User;
+use App\Activity;
+use Illuminate\Support\Facades\Auth;
 
 class RelationshipController extends Controller
 {
     public function follow(User $profile_user, User $user)
     {
         $relation = new Relationship;
-        $relation->create([
+        $relation = $relation->create([
             'follower_id' => $user->id, 
             'followed_id' => $profile_user->id
         ]);
+
+        //Storing activity data to activity model.
+        $activity = new Activity;
+        $activity->create([
+            'user_id' => Auth::id(),
+            'action_id' => $relation->id,
+            'action_type' => 'followed'
+        ]);
         return back();
     }
 
     public function unfollow(User $profile_user, User $user)
     {
-        Relationship::where('followed_id', $profile_user->id)->where('follower_id', $user->id)->delete();
+        $relation = Relationship::where('followed_id', $profile_user->id)->where('follower_id', $user->id)->first();
```

Comment:
you can make this short by just using the Relationship model. Please find it modular